### PR TITLE
Implement the Delete Statement using vectorization

### LINKED ISSUE #1342: Implement the Delete Statement using vectorization
Now, the Delete Statement in BE is not vectorized.
It's the last operation which is not vectorized.
After the operation is vectorized, all non-vectorized code paths can be removed.

### PR DESCRIPTION
**Pull request by chaoyli**
The vectorized PushHandler is used to vectorize SparkLoad.
And also, implement the Delete Statement in a vectorized form.

### CODE REVIEW

File: be/src/storage/vectorized/push_handler.cpp [hunk 1]
```diff
@@ -410,8 +441,68 @@ void PushHandler::_get_tablet_infos(const std::vector<TabletVars>& tablet_vars,
...
+
+        // 2. init RowsetBuilder of cur_tablet for current push
+        RowsetWriterContext context(kDataFormatUnknown, config::storage_format_version);
+        context.rowset_id = StorageEngine::instance()->next_rowset_id();
+        context.tablet_uid = cur_tablet->tablet_uid();
+        context.tablet_id = cur_tablet->tablet_id();
+        context.partition_id = _request.partition_id;
+        context.tablet_schema_hash = cur_tablet->schema_hash();
+        context.rowset_type = BETA_ROWSET;
+        context.rowset_path_prefix = cur_tablet->tablet_path();
```
Comment (wyb) [inline be/src/storage/vectorized/push_handler.cpp:463]: context.rowset_path_prefix = cur_tablet->schema_hash_path();

File: be/src/storage/vectorized/push_handler.h [hunk 2]
```diff
@@ -78,8 +78,8 @@ class PushHandler {
 
     void _get_tablet_infos(const std::vector<TabletVars>& tablet_infos, std::vector<TTabletInfo>* tablet_info_vec);
 
-    Status _convert(const TabletSharedPtr& cur_tablet, const TabletSharedPtr& new_tablet_vec,
-                    RowsetSharedPtr* cur_rowset, RowsetSharedPtr* new_rowset);
+    Status _convert(const TabletSharedPtr& cur_tablet, RowsetSharedPtr* cur_rowset);
+    Status _convert_v2(const TabletSharedPtr& cur_tablet, RowsetSharedPtr* cur_rowset);
```
Comment (wyb) [inline be/src/storage/vectorized/push_handler.h:82]: _load_convert
_delete_convert?

**Review by wyb — DISMISSED** [review]

**Comment by chaoyli** [comment]
run starrocks_be_unittest

**Review by wyb — DISMISSED** [review]

File: be/src/storage/task/engine_batch_load_task.cpp [hunk 3]
```diff
@@ -227,11 +227,15 @@ OLAPStatus EngineBatchLoadTask::_delete_data(const TPushReq& request, std::vecto
     }
 
     // 2. Process delete data by push interface
-    PushHandler push_handler;
-    if (request.__isset.transaction_id) {
-        res = push_handler.process_streaming_ingestion(tablet, request, PUSH_FOR_DELETE, tablet_info_vec);
-    } else {
-        res = OLAP_ERR_PUSH_BATCH_PROCESS_REMOVED;
+    bool use_vectorized = request.__isset.use_vectorized && request.use_vectorized;
```
Comment (sduzh) [inline be/src/storage/task/engine_batch_load_task.cpp:230]: Unused variable

File: fe/fe-core/src/main/java/com/starrocks/task/PushTask.java [hunk 4]
```diff
@@ -91,7 +91,7 @@ public PushTask(TResourceInfo resourceInfo, long backendId, long dbId, long tabl
         this.transactionId = transactionId;
         this.tBrokerScanRange = null;
         this.tDescriptorTable = null;
-        this.useVectorized = false;
+        this.useVectorized = true;
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/task/PushTask.java:94]: Unused variable

Comment (chaoyli) [reply fe/fe-core/src/main/java/com/starrocks/task/PushTask.java:94]: PushTask is also used by SparkLoad, this is meaningful

**Review by sduzh — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**